packages/android-lib/src/main/kotlin/com/outsystems/plugins/osgeolocation/controller/OSGLOCServiceHelper.kt

```diff
@@ -0,0 +1,165 @@
+     * @return true if location was checked and is on, false if it requires user to resolve issue (e.g. turn on location)
+     *          If false, the result is returned in `resolveLocationSettingsResultFlow`
+     * @throws [OSGLOCException.OSGLOCSettingsException] if an error occurs that is not resolvable by user
+     */
+    internal suspend fun checkLocationSettings(
+        activity: Activity,
+        options: OSGLOCLocationOptions,
+        interval: Long
+    ): Boolean {
+
+        val request = LocationRequest.Builder(
+            if (options.enableHighAccuracy) Priority.PRIORITY_HIGH_ACCURACY else Priority.PRIORITY_BALANCED_POWER_ACCURACY,
+            interval
+        ).build()
+
+        val builder = LocationSettingsRequest.Builder()
+        builder.addLocationRequest(request)
+        val client = LocationServices.getSettingsClient(activity)
+
+        try {
+            client.checkLocationSettings(builder.build()).await()
+            return true
+        } catch (e: ResolvableApiException) {
+
+            // Show the dialog to enable location by calling startResolutionForResult(),
+            // and then handle the result in onActivityResult
+            val resolutionBuilder: IntentSenderRequest.Builder =
+                IntentSenderRequest.Builder(e.resolution)
+            val resolution: IntentSenderRequest = resolutionBuilder.build()
+
+            activityLauncher.launch(resolution)
+        } catch (e: Exception) {
+            throw OSGLOCException.OSGLOCSettingsException(
+                message = "There is an error with the location settings.",
+                cause = e
+            )
+        }
+        return false
+    }
+
+    /**
+     * Checks if the device has google play services, required to use [FusedLocationProviderClient]
+     * @param activity the Android activity from which the location request is being triggered
+     * @return true if google play services is available, false otherwise
+     */
+    internal fun checkGooglePlayServicesAvailable(activity: Activity): Result<Unit> {
+        val googleApiAvailability = GoogleApiAvailability.getInstance()
+        val status = googleApiAvailability.isGooglePlayServicesAvailable(activity)
+
+        return if (status != ConnectionResult.SUCCESS) {
+            if (googleApiAvailability.isUserResolvableError(status)) {
+                googleApiAvailability.getErrorDialog(activity, status, 1)?.show()
+                Result.failure(
+                    OSGLOCException.OSGLOCGoogleServicesException(
+                        resolvable = true,
+                        message = "Google Play Services error user resolvable."
+                    )
+                )
+            } else {
+                Result.failure(
+                    OSGLOCException.OSGLOCGoogleServicesException(
+                        resolvable = false,
+                        message = "Google Play Services error."
+                    )
+                )
```

Comment:
Maybe we could avoid duplication of the `Result.failure(...)`.

Reply:
Refactored, can you check?